spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/CassandraCatalog.scala

```diff
@@ -53,7 +37,7 @@ private[cassandra] class CassandraCatalog(cc: CassandraSQLContext) extends Catal
 
   override def tableExists(tableIdentifier: Seq[String]): Boolean = {
     val (cluster, database, table) = getClusterDBTableNames(tableIdentifier)
-    val schema = schemas.get(cluster)
+    val schema = cc.getCassandraSchema(cluster)
     val tabDef = for (ksDef <- schema.keyspaceByName.get(database); tabDef <- ksDef.tableByName.get(table)) yield tabDef
     tabDef.isDefined
   }
```

Comment:
It was left in previous integration, for we don't have local cache for the tables, all tables are construct on the fly. I will open a new ticket to address all of them.